### ISSUE TITLE
Improve features for restarting from checkpoints

### ISSUE THREAD

**Issue by jackapbutler** [OpenBioML/chemnlp#273]
In order to ensure we can rescue runs that crash in the middle of epochs we need to improve the quality of our checkpointing and restarting mechanisms
- See [here](https://discuss.huggingface.co/t/how-to-continue-training-and-not-overwrite-checkpoint-number/25370) for how models are saved when continuing (we’ll overwrite checkpoints). We already have something implemented but need to decide how we handle restarting in general e.g. need to consider learning rate scheduler, which data we have already trained on etc… if we do this.

Two options: use `from_pretrained` or `resume_from_checkpoint` in `Trainer.train()` maybe including a `uuid` for each run.
    - how we efficiently restart on a dataset?
    - how to save down dataloader state with HF?
    - consider [here](https://github.com/huggingface/transformers/blob/v4.27.2/src/transformers/trainer_pt_utils.py#L1022) and [here](https://huggingface.co/docs/accelerate/usage_guides/checkpoint) for more information
